drivers/eeprom/eeprom_rv3028/eeprom_rv3028.c

```diff
@@ -0,0 +1,179 @@
+		err = rv3028_read_reg8(rtc_dev, RV3028_REG_EEPROM_DATA, &dst[i]);
+		if (err) {
+			goto unlock;
+		}
+	}
+
+unlock:
+	rv3028_exit_eerd(rtc_dev);
+	rv3028_unlock_sem(rtc_dev);
+
+	return err;
+}
+
+static int rv3028_eeprom_write(const struct device *dev, off_t offset, const void *data, size_t len)
+{
+	const struct rv3028_eeprom_config *config = dev->config;
+	const struct device *rtc_dev = config->parent;
+	const uint8_t *src = data;
+	uint8_t cmd[2];
+	int err;
+
+	if (!len) {
+		return 0;
+	}
+
+	if (offset + len > config->size) {
+		LOG_ERR("attempt to write past device boundary");
+		return -EINVAL;
+	}
+
+	rv3028_lock_sem(rtc_dev);
+
+	err = rv3028_enter_eerd(rtc_dev);
+	if (err) {
+		goto unlock;
+	}
+
+	for (size_t i = 0; i < len; i++) {
+		cmd[0] = config->addr + offset + i;
+		cmd[1] = src[i];
+
+		err = rv3028_write_regs(rtc_dev, RV3028_REG_EEPROM_ADDRESS, cmd, sizeof(cmd));
+		if (err) {
+			goto unlock;
+		}
+
+		err = rv3028_eeprom_command(rtc_dev, RV3028_EEPROM_CMD_WRITE);
+		if (err) {
+			goto unlock;
+		}
+
+		k_msleep(RV3028_EEBUSY_WRITE_POLL_MS);
+
+		err = rv3028_eeprom_wait_busy(rtc_dev, RV3028_EEBUSY_WRITE_POLL_MS);
+		if (err) {
+			goto unlock;
+		}
+	}
+
+unlock:
+	rv3028_exit_eerd(rtc_dev);
+	rv3028_unlock_sem(rtc_dev);
+
+	return err;
+}
+
+static int rv3028_eeprom_init(const struct device *dev)
+{
+	const struct rv3028_eeprom_config *config = dev->config;
+
+	if (!device_is_ready(config->parent)) {
+		LOG_ERR("parent device %s is not ready", config->parent->name);
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static const struct eeprom_driver_api rv3028_eeprom_api = {
+	.size = rv3028_eeprom_get_size,
+	.read = rv3028_eeprom_read,
+	.write = rv3028_eeprom_write,
+};
+
+#define RV3028_EEPROM_ASSERT_AREA_SIZE(inst)                                                       \
+	BUILD_ASSERT(DT_INST_REG_SIZE(inst) + DT_INST_REG_ADDR(inst) <= RV3028_USER_EEPROM_SIZE,   \
+		     "Invalid RV3028 EEPROM area size");
+
+#define RV3028_EEPROM_INIT(inst)                                                                   \
+	RV3028_EEPROM_ASSERT_AREA_SIZE(inst)                                                       \
+                                                                                                   \
+	static const struct rv3028_eeprom_config rv3028_eeprom_config_##inst = {                   \
+		.parent = DEVICE_DT_GET(DT_INST_BUS(inst)),                                        \
+		.addr = DT_INST_REG_ADDR(inst),                                                    \
+		.size = DT_INST_REG_SIZE(inst),                                                    \
+	};                                                                                         \
+                                                                                                   \
+	DEVICE_DT_INST_DEFINE(inst, rv3028_eeprom_init, NULL, NULL, &rv3028_eeprom_config_##inst,  \
+			      POST_KERNEL, CONFIG_EEPROM_RV3028_INIT_PRIORITY,                     \
```

Comment:
temping to suggest making this `CONFIG_RTC_INIT_PRIORITY`, the option is going to be available because you depend on the driver that depends on RTC, if you initialize both the parent and child at the same priority then the subpriority stuff will ensure that they are initialized in the correct sequence and you'd save one Kconfig option to mess with.

Con is that it looks odd to have an eeprom device with an rtc priority, but then it already a custom symbol so why not.

What do you think?

Reply:
Sounds reasonable. Thanks for the suggestion.